app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -387,13 +389,42 @@ class LocalFileLocator(private val applicationFilesDir: String, private val reso
 }
 
 class DeviceDimensions {
-    private var width = 720
-    private var height = 1480
+    private var height = 720
+    private var width = 1480
 
-    fun getDeviceDimensions(windowManager: WindowManager, displayMetrics: DisplayMetrics) {
-        windowManager.defaultDisplay.getMetrics(displayMetrics)
-        width = displayMetrics.widthPixels
+    companion object {
+        const val portrait = "SCREEN_ORIENTATION_PORTRAIT"
+        const val landscape = "SCREEN_ORIENTATION_LANDSCAPE"
+    }
+
+    fun getDeviceDimensions(windowManager: WindowManager, displayMetrics: DisplayMetrics, orientation: Int) {
+        val navBarSize = getNavigationBarSize(windowManager)
+        windowManager.defaultDisplay.getRealMetrics(displayMetrics)
         height = displayMetrics.heightPixels
+        width = displayMetrics.widthPixels
+        windowManager.defaultDisplay.getMetrics(displayMetrics)
+
+        when (checkOrientation(orientation)) {
+            portrait -> if (navBarSize.y > 0) height += navBarSize.y
+            landscape -> if (navBarSize.x > 0) width += navBarSize.x
+        }
+    }
+
+    private fun getNavigationBarSize(windowManager: WindowManager): Point {
+        val appUsableSize = Point()
+        val realScreenSize = Point()
+        val display = windowManager.defaultDisplay
+        display.getSize(appUsableSize)
+        display.getRealSize(realScreenSize)
+
+        return Point(realScreenSize.x - appUsableSize.x, realScreenSize.y - appUsableSize.y)
+    }
+
+    private fun checkOrientation(orientation: Int): String {
```

Comment:
I think this function could be removed, since the check above looks so similar.

Reply:
Alternatively, it would be more functional if you left the `get` language and had the function return the geometry as needed, so that there wasn't intermediate state left around.

Reply:
Removed this function